SignalMessaging/environment/AppSetup.m

```diff
@@ -46,7 +51,18 @@ + (void)setupEnvironment:(CallMessageHandlerBlock)callMessageHandlerBlock
         [NSKeyedUnarchiver setClass:[OWSUserProfile class] forClassName:[OWSUserProfile collection]];
         [NSKeyedUnarchiver setClass:[OWSDatabaseMigration class] forClassName:[OWSDatabaseMigration collection]];
 
-        [OWSStorage setupStorage];
+        [OWSStorage setupStorageWithMigrationBlock:^() {
+            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+                // Don't start database migrations until storage is ready.
+                [VersionMigrations performUpdateCheckWithCompletion:^() {
```

Comment:
Hmmm... looks like this used to get called on the main thread. 

Are you backgrounding it because we don't want `runAllOutstandingWithCompletion` to block the main thread? That seems reasonable.

Since we're comparing state in `performUpdateCheckWithCompletion` (e.g. `AppVersion.instance.lastAppVersion` and friends), maybe it makes sense to continue to call `performUpdateCheckWithCompletion` on the main thread, but have `performUpdateCheckWithCompletion` do the background dispatch for the potentially blocking parts: `assumeAllExistingMigrationsRun`/`runAllOutstandingWithCompletion`

In practice, it looks like all the state we're accessing is thread safe anyway, so I don't see a problem with the change as written, just nitting over style. Take it or leave it.

Reply:
Okay, moved this "dispatch off main thread" to just cover `runAllOutstandingWithCompletion`.  `assumeAllExistingMigrationsRun` is cheap enough not to matter.